sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -997,3 +1164,196 @@ def _from_generated(cls, score):
+        task metadata.
+
+    :ivar str name: The name of the task.
+    :ivar results: The results of the analysis.
+    :vartype results: list[~azure.ai.textanalytics.ExtractKeyPhrasesResult]
+    """
+
+    def __init__(self, **kwargs):
+        self.name = kwargs.get("name", None)
+        self.results = kwargs.get("results", [])
+
+    def __repr__(self, **kwargs):
+        return "KeyPhraseExtractionTaskResult(name={}, results={})" \
+            .format(self.name, repr(self.results))[:1024]
+
+
+class TextAnalysisResult(DictMixin):
+    """TextAnalysisResult contains the results of multiple text analyses performed on a batch of documents.
+
+    :ivar entities_recognition_results: A list of objects containing results for all Entity Recognition tasks
+        included in the analysis.
+    :vartype entities_recognition_results: list[~azure.ai.textanalytics.EntitiesRecognitionTaskResult]
+    :ivar pii_entities_recognition_results: A list of objects containing results for all PII Entity Recognition
+        tasks included in the analysis.
+    :vartype pii_entities_recogition_results: list[~azure.ai.textanalytics.PiiEntitiesRecognitionTaskResult]
+    :ivar key_phrase_extraction_results: A list of objects containing results for all Key Phrase Extraction tasks 
+        included in the analysis.
+    :vartype key_phrase_extraction_results: list[~azure.ai.textanalytics.KeyPhraseExtractionTaskResult]
+    """
+    def __init__(self, **kwargs):
+        self.entities_recognition_results = kwargs.get("entities_recognition_results", [])
+        self.pii_entities_recognition_results = kwargs.get("pii_entities_recognition_results", [])
+        self.key_phrase_extraction_results = kwargs.get("key_phrase_extraction_results", [])
+
+    def __repr__(self):
+        return "TextAnalysisResult(entities_recognition_results={}, pii_entities_recognition_results={}, \
+            key_phrase_extraction_results={})" \
+            .format(
+                repr(self.entities_recognition_results),
+                repr(self.pii_entities_recognition_results),
+                repr(self.key_phrase_extraction_results)
+            )[:1024]
+
+
+class RequestStatistics(DictMixin):
+    def __init__(self, **kwargs):
+        self.documents_count = kwargs.get("documents_count")
+        self.valid_documents_count = kwargs.get("valid_documents_count")
+        self.erroneous_documents_count = kwargs.get("erroneous_documents_count")
+        self.transactions_count = kwargs.get("transactions_count")
+
+    @classmethod
+    def _from_generated(cls, request_statistics):
+        return cls(
+            documents_count=request_statistics.documents_count,
+            valid_documents_count=request_statistics.valid_documents_count,
+            erroneous_documents_count=request_statistics.erroneous_documents_count,
+            transactions_count=request_statistics.transactions_count
+        )
+
+    def __repr__(self, **kwargs):  # TODO: why is this never called?
```

Comment:
This `repr`'s only called when you try to represent the class as a string

Reply:
Ok, thanks for the info!  I removed this comment for now, I'm not sure if the issue I was having is still present or not.  I think I was debugging something one day and not seeing the proper representation at the time, but haven't noticed an issue with it since.